libvast/native-plugins/rebuild.cpp

```diff
@@ -11,384 +11,618 @@
-                    num_total, num_transforming, num_total - num_transformed)});
-    indicator->tick();
+  /// Start a new rebuild.
+  auto start(start_options options) -> caf::result<void> {
+    if (options.parallel == 0)
+      return caf::make_error(ec::invalid_configuration,
+                             "rebuild requires a non-zero parallel level");
+    if (options.undersized)
+      options.all = true;
+    if (options.automatic && run)
+      return {};
+    if (run && !run->options.automatic)
+      return caf::make_error(
+        ec::invalid_argument,
+        fmt::format("{} refuses to start rebuild while a rebuild is still "
+                    "ongoing ({}/{} done); consider running 'vast rebuild "
+                    "stop'",
+                    *self, run->statistics.num_completed,
+                    run->statistics.num_total));
+    if (!options.automatic && run && run->options.automatic) {
+      auto rp = self->make_response_promise<void>();
+      self
+        ->request(static_cast<rebuilder_actor>(self), caf::infinite,
+                  atom::stop_v, stop_options{.detached = false})
+        .then(
+          [this, rp, options = std::move(options)]() mutable {
+            rp.delegate(static_cast<rebuilder_actor>(self), atom::start_v,
+                        std::move(options));
+          },
+          [rp](caf::error& err) mutable {
+            rp.deliver(std::move(err));
+          });
+      return rp;
+    }
+    run.emplace();
+    run->options = std::move(options);
+    const auto lookup_id = uuid::random();
+    const auto max_partition_version = run->options.all
+                                         ? version::partition_version
+                                         : version::partition_version - 1;
+    VAST_DEBUG("{} requests {}{} partitions matching the expression {}", *self,
+               run->options.all ? "all" : "outdated",
+               run->options.undersized ? " undersized" : "",
+               run->options.expression);
+    auto rp = self->make_response_promise<void>();
+    auto finish = [this, rp](caf::error err, bool silent = false) mutable {
+      if (!silent) {
+        if (run->statistics.num_completed == 0)
+          VAST_INFO("{} had nothing to do", *self);
+        else
+          VAST_INFO("{} rebuilt {} into {} partitions", *self,
+                    run->statistics.num_completed, run->statistics.num_results);
+      }
+      run.reset();
+      emit_telemetry();
+      if (run->options.detached)
+        return;
+      if (err) {
+        rp.deliver(std::move(err));
+        return;
+      }
+      static_cast<caf::response_promise&>(rp).deliver(caf::unit);
+    };
+    if (run->options.detached)
+      static_cast<caf::response_promise&>(rp).deliver(caf::unit);
+    self
+      ->request(catalog, caf::infinite, atom::candidates_v, lookup_id,
+                run->options.expression, max_partition_version)
```

Comment:
This should get an option not to generate metrics from the catalog. Looking at our dashboard, we have lots of catalog metrics that yield all results, so the graph with the candidate partitions is unusable now.

Reply:
Per discussion with @lava this morning in Slack I added an "issuer" field to query metrics metadata. This allows for disambiguating in the dashboard.